src/main/java/com/philips/swcoe/cerberus/hounds/JavaCodeMetricsWithDiff.java

```diff
@@ -61,15 +61,23 @@ public Integer call() throws Exception {
             this.validateFilePath(methodConfigFile);
             methodConfig = Files.readAllLines(new File(methodConfigFile).toPath(), Charset.defaultCharset());
         }
+        if("md".equalsIgnoreCase(reportFormat) || "html".equalsIgnoreCase(reportFormat)) {
+            this.validateReportFormat(metricFormat);
+        }
         CodeMetricsDiffService codeMetricsDiffService = new CodeMetricsDiffService(pathToPrevious, pathToCurrent);
         List<CodeMetricsClassResult> metricsResult = codeMetricsDiffService.getMetricsFromSourceCode();
-        char delimiter =  DelimeterForReport.valueOf(reportFormat.toUpperCase()).getDelimeter();
         if("vertical".equalsIgnoreCase(metricFormat)) {
-            this.writeToUI(new CodeMetricsVerticalWriterService(classConfig, methodConfig, delimiter).generateMetricsReport(metricsResult));
+            this.writeToUI(new CodeMetricsVerticalWriterService(classConfig, methodConfig, reportFormat.toUpperCase()).generateMetricsReport(metricsResult));
         } else {
-            this.writeToUI(new CodeMetricsHorizontalWriterService(classConfig, methodConfig, delimiter).generateMetricsReport(metricsResult));
+            this.writeToUI(new CodeMetricsHorizontalWriterService(classConfig, methodConfig, reportFormat.toUpperCase()).generateMetricsReport(metricsResult));
         }
         return 0;
     }
+
+    private void validateReportFormat(String metricFormat) {
+        if("horizontal".equalsIgnoreCase(metricFormat)) {
+            throw new CommandLine.ParameterException(spec.commandLine(), INVALID_REPORT_FORMAT);
```

Comment:
We should not, because defaulting to certain behaviour allows cultivation of bad usage and puts pressure on us in future when we want to do some changes. Keeping less default behaviour helps us to keep the design evolutionary by nature. 

For eg.. I default to CSV here, and tomm some thing else comes in for horizontal report, we get into soup where we cannot do this change with out causing the break in usage. for eg ... if some one has scripted a shell script with these parameters and they think that we are anyway defaulting to CSV and post doing a new release we should also tell them to update explicitly and they might even revolt back saying why can't we continue defaulting to CSV which is waste. 

Embedding default behaviours in the software, is a kind of an antipattern which needs to be avoided as much as possible to prevent future wastes, and keep things as evolutionary as possible.